src/components/shared/searchView/SearchResults.tsx

```diff
@@ -71,16 +73,20 @@ class SearchResults extends React.Component<ISearchResultsProps, ISearchResultsS
     }
 
     private fetchAllLines = async () => {
-        const lines = await LineService.fetchAllLines();
-        if (lines !== null) {
+        try {
+            const lines = await LineService.fetchAllLines();
```

Comment:
Hmm, now if lines is empty array, it will not throw error.

Maybe we need something like:
const lines = await LineService.fetchAllLines();
if (lines.length === 0) throw new Error();

Reply:
Why would we throw an error if fetchAllLines return empty array?
fetchAllLines should throw an error if it has problems fetching something, it should never return empty array if an error occurs. Only time it should return empty array is if the database is empty, and in that case, it should not throw any error